### ISSUE TITLE
Cannot build formeo

### ISSUE THREAD

**Issue by AlexBreskin** [Draggable/formeo#173]
Hi there,

I've been attempting to build Formeo on a Windows 10 machine using Command Prompt running under Administrator mode, but have been unable to up to this point.

The commands used so far are:

> git clone https://github.com/Draggable/formeo.git
> cd formeo
> npm install && npm run dev

This process will run up until it hits the following errors:

> npm WARN prepublish-on-install As of npm@5, `prepublish` scripts are deprecated.
> npm WARN prepublish-on-install Use `prepare` for build steps and `prepublishOnly` for upload-only.
> npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.
> 
> > mini-css-extract-plugin@0.5.0 prepublish E:\Dev\Recognite\Dev\formeo\node_modules\mini-css-extract-plugin
> > npm run build
> 
> 
> > mini-css-extract-plugin@0.5.0 prebuild E:\Dev\Recognite\Dev\formeo\node_modules\mini-css-extract-plugin
> > npm run clean
> 
> 
> > mini-css-extract-plugin@0.5.0 clean E:\Dev\Recognite\Dev\formeo\node_modules\mini-css-extract-plugin
> > del-cli dist
> 
> 
> > mini-css-extract-plugin@0.5.0 build E:\Dev\Recognite\Dev\formeo\node_modules\mini-css-extract-plugin
> > cross-env NODE_ENV=production babel src -d dist --ignore 'src/**/*.test.js' --copy-files
> 
> src doesn't exist
> npm ERR! code ELIFECYCLE
> npm ERR! errno 2
> npm ERR! mini-css-extract-plugin@0.5.0 build: `cross-env NODE_ENV=production babel src -d dist --ignore 'src/**/*.test.js' --copy-files`
> npm ERR! Exit status 2
> npm ERR!
> npm ERR! Failed at the mini-css-extract-plugin@0.5.0 build script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     C:\Users\user\AppData\Roaming\npm-cache\_logs\2019-02-18T00_44_54_997Z-debug.log
> npm ERR! code ELIFECYCLE
> npm ERR! errno 2
> npm ERR! mini-css-extract-plugin@0.5.0 prepublish: `npm run build`
> npm ERR! Exit status 2
> npm ERR!
> npm ERR! Failed at the mini-css-extract-plugin@0.5.0 prepublish script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     C:\Users\user\AppData\Roaming\npm-cache\_logs\2019-02-18T00_44_55_043Z-debug.log
> npm ERR! code ELIFECYCLE
> npm ERR! errno 2
> npm ERR! formeo@1.0.0-beta.1 postinstall: `npm --prefix ./node_modules/mini-css-extract-plugin install ./node_modules/mini-css-extract-plugin && npm run --prefix ./node_modules/mini-css-extract-plugin build`
> npm ERR! Exit status 2
> npm ERR!
> npm ERR! Failed at the formeo@1.0.0-beta.1 postinstall script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     C:\Users\user\AppData\Roaming\npm-cache\_logs\2019-02-18T00_44_55_146Z-debug.log

Any help here would be greatly appreciated. I have attempted to use webpack -v, but it instead goes through a process of installing webpack-cli and fails along the way, and subsequent attempts to reinstall still fail as mentioned previously.

**Comment by charmis** [comment]
Make sure you have installed yarn. Please refer [yarn install page](https://yarnpkg.com/lang/en/docs/install)

Execute the following commands one by one:

```
git clone https://github.com/Draggable/formeo.git
cd formeo
yarn install
yarn run build
```
Once all the above commands are done, you would be able to see a folder `formeo\demo` which has `index.html` and an `assets` folder.

Open the index.html in browser and you would be good to see the **formeo** designer screen.

**Comment by AlexBreskin** [comment]
Thanks @charmis 

This did the trick! I'm now able to build the application following those steps.

Would it be possible to refer to these steps in the documentation for future users?

**Comment by charmis** [comment]
You are welcome @AlexBreskinRecognite 

I am not exactly sure how to fix when using `npm`. 

What I found is that the npm script is looking for `src` folder and contents inside `\formeo\node_modules\mini-css-extract-plugin` which does not exist. But while using `yarn`, the `src` and its contents exist.

@kevinchappell Do you think it would be a good idea to add these steps in the documentation?

**Comment by kevinchappell** [comment]
@charmis absolutely. I took a look last night and the steps need to be updated as well as fixing install when using npm.

**Comment by intelligence0504** [comment]
when I open the index.html in browser, I have some issues.

Fetch API cannot load file:///home/david/Workspace/formeo-master/formeo/demo/assets/img/formeo-sprite.svg. URL scheme must be "http" or "https" for CORS request.
(anonymous) @ demo.min.js:28
demo.min.js:11 Access to XMLHttpRequest at 'file:///home/david/Workspace/formeo-master/formeo/demo/assets/lang/en-US.lang' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, https.
(anonymous) @ demo.min.js:11
index.html:1 Uncaught (in promise) Promise

**Comment by kevinchappell** [comment]
@intelligence0504 this is a separate issue. formeo uses fetch to load icon and language resources which does not work for local files. You'll have to serve index.html, you can do this using the built-in dev server or if you want to test the production build you can use an npm module like [serve](https://www.npmjs.com/package/serve)

try the following:

```bash
npm i -g serve
cd home/david/Workspace/formeo-master/formeo/
npm install
npm run build
serve demo/
```

This will install the `serve` module, change directory to formeo, install its dependencies, build and serve the demo/ directory. If this is for development purposes I recommend `npm start` to run the webpack dev server instead.


also closing this issue because the original problem has been solved by #174

**Comment by intelligence0504** [comment]
Thanks...Do you have react app using forme?

**Comment by intelligence0504** [comment]
I don't know the way of usage.

**Comment by kevinchappell** [comment]
@intelligence0504 not yet though i think others have.

**Comment by intelligence0504** [comment]
Thanks for your answer.
I have made custom field (image) at react app using jquery form builder.
But I cant display custom field on preview Popup.
And how should I implement section(function like row and column on formeo) in jquery form builder?
Help me.


Sent from Mail for Windows 10

**Comment by intelligence0504** [comment]
Does Formeo supply French in i18n?
I can not see French   on https://draggable.github.io/formeo/ .

Sent from Mail for Windows 10

**Comment by intelligence0504** [comment]
Hello.. How are you?
I cant install formeo using npm.
When I run “npm I formeo” , got some issue.
Help me. 
Thx.
 Sent from Mail for Windows 10
